packages/react/src/components/UIShell/components/SideNav.tsx

```diff
@@ -34,6 +34,7 @@ export enum SIDE_NAV_TYPE {
   DEFAULT = 'default',
   RAIL = 'rail',
   PANEL = 'panel',
+  TREEVIEW = 'treeview',
```

Comment:
I think we briefly discussed this during a dev sync but does the user need to set the navType to treeview when there are nested menus or can that be automatically set?

Reply:
For the moment it needs to be automatically set, but we'd have to discuss about the requirements for the automatic setting. Mainly, would the existence of a nested menu mean that the whatever `navType` will be overridden? 

cc: @alisonjoseph @oliviaflory

Reply:
yeah I think having it tied to the role makes sense and we can revisit this in a future PR :+1: